x-pack/plugins/apm/server/lib/apm_telemetry/types.ts

```diff
@@ -24,6 +24,47 @@ export interface AggregatedTransactionsCounts {
   transaction_count: number;
 }
 
+export interface APMPerService {
+  service_id: string;
+  timed_out: boolean;
+  cloud: {
+    availability_zones: string[];
+    regions: string[];
+    providers: string[];
+  };
+  faas: {
+    trigger: {
+      type: string[];
+    };
+  };
+  agent: {
+    name: string | number | null;
+    version: string | number | null;
```

Comment:
I think you need to drop the union types here below:

```suggestion
    name: string;
    version: string;
```

Reply:
vscode really doesn't like that but it did fix the schema generation!